Added documentation for settings limiting total shards and primary shards per node at both cluster and index level. (#9791)
opensearch-project/documentation-website: _install-and-configure/configuring-opensearch/index-settings.md

- `index.optimize_doc_id_lookup.fuzzy_set.false_positive_probability` (Double): Sets the false-positive probability for the underlying `fuzzy_set` (that is, the Bloom filter). A lower false-positive probability ensures higher throughput for upsert and get operations but results in increased storage and memory use. Allowed values range between `0.01` and `0.50`. Default is `0.20`.

> - `index.routing.allocation.total_shards_per_node` (Integer): Maximum number of primary and replica shards (total) from a single index that can be allocated on a single node. Defaults to `-1` (unlimited). Helps control per-index shard distribution across nodes by limiting shards per node. Use with caution as shards from this index may remain unallocated if nodes reach their configured limits.

Reviewer: ```suggestion
- `index.routing.allocation.total_shards_per_node` (Integer): The maximum combined total number of primary and replica shards from a single index that can be allocated to a single node. Default is `-1` (unlimited). Helps control per-index shard distribution across nodes by limiting the number of shards per node. Use with caution because shards from this index may remain unallocated if nodes reach their configured limits.
```

> - `index.routing.allocation.total_primary_shards_per_node` (Integer): Maximum number of primary shards from a single index that can be allocated on a single node. This setting is applicable only for remote-backed clusters. Defaults to `-1` (unlimited). Helps control per-index primary shard distribution across nodes by limiting primary shards per node. Use with caution as primary shards from this index may remain unallocated if nodes reach their configured limits.

Reviewer: ```suggestion
- `index.routing.allocation.total_primary_shards_per_node` (Integer): The maximum number of primary shards from a single index that can be allocated to a single node. This setting is applicable only for remote-backed clusters. Default is `-1` (unlimited). Helps control per-index primary shard distribution across nodes by limiting the number of primary shards per node. Use with caution because primary shards from this index may remain unallocated if nodes reach their configured limits.
```
